test/testdrive/source-format-json.td

```diff
@@ -0,0 +1,50 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Verify behavior of FORMAT JSON
+
+$ kafka-create-topic topic=data partitions=1
+$ kafka-ingest format=bytes topic=data
+{"a":"b","c":"d"}
+
+> CREATE CONNECTION kafka_conn
+  TO KAFKA (BROKER '${testdrive.kafka-addr}');
+
+> CREATE SOURCE data
+  FROM KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-data-${testdrive.seed}')
+  FORMAT JSON;
+
+> SELECT DISTINCT pg_typeof(data) FROM data;
+jsonb
+
+> SELECT * FROM data
+"{\"a\":\"b\",\"c\":\"d\"}"
+
+# It's a dict so this is not just a string masquerading as JSON
+> SELECT data -> 'a' FROM data;
+"\"b\""
+
+$ kafka-ingest format=bytes topic=data
+[1,2,3]
+1
+1.23
+"hello"
+
+> SELECT * FROM data
+[1,2,3]
+1
+1.23
+"\"hello\""
+"{\"a\":\"b\",\"c\":\"d\"}"
+
+$ kafka-ingest format=bytes topic=data
+hello
+
+!SELECT * FROM data
+exact:Decode error: Bytes: Failed to decode JSON (original bytes: [68, 65, 6c, 6c, 6f])
```

Comment:
The error message is not very informative -- can you print the bytes as a string?

Also, this has the potential of generating an extremely long error message. Maybe it would be better to print just the offset of the error within the string + some context?

Reply:
Tried to improve this as much as was reasonable; if it is something that can be decoded as a string, we will do that now.